blockchain/chain_sync/src/sync.rs

```diff
@@ -68,7 +68,7 @@ pub struct ChainSyncer<DB, TBeacon> {
     next_sync_target: SyncBucket,
 
     /// access and store tipsets / blocks / messages
-    chain_store: ChainStore<DB>,
+    chain_store: Arc<RwLock<ChainStore<DB>>>,
```

Comment:
is it possible to just wrap what is needed in an Arc<RwLock<..>> instead of wrapping the whole chain store in a mutex? This kinda worries me

Reply:
I thought about that but I think it makes sense in the context of both places being able to reflect the changes that are coming in from both sides. Like unless we go through the route of generating a new chain_store per operation which might take a lot of restructuring. Example things like HeadChange is something that would need to be reflected from both sides.

Additionally, I thought it was okay because state is behind one and trying to circumvent by using module level functions, getting the block_store from the state_manager and using sync types of the async library. 

I will look closer and see if we can just get what we need through

Reply:
I just mean wrapping the variables other than the blockstore in the mutex, so you don't wrap the whole struct (and would restrict blockstore access even if just one of the other variables are locked)